quic/s2n-quic-transport/src/recovery/manager.rs

```diff
@@ -304,6 +305,39 @@ impl<Config: endpoint::Config> Manager<Config> {
         self.pto.on_transmit(context)
     }
 
+    /// Process pending ACK information.
+    ///
+    /// Update congestion controller, timers and meta data around acked packet ranges.
+    pub fn on_process_pending_ack_ranges<Ctx: Context<Config>, Pub: event::ConnectionPublisher>(
+        &mut self,
+        timestamp: Timestamp,
+        pending_ack_ranges: &mut PendingAckRanges,
+        context: &mut Ctx,
+        publisher: &mut Pub,
+    ) -> Result<(), transport::Error> {
+        debug_assert!(
+            !pending_ack_ranges.is_empty(),
+            "pending_ack_ranges should be non-empty since connection indicated ack interest"
+        );
+
+        let largest_acked_packet_number = pending_ack_ranges
+            .max_value()
+            .expect("pending range should not be empty");
+        self.process_acks(
+            timestamp,
+            pending_ack_ranges
+                .iter()
+                .map(|ack_range| PacketNumberRange::new(*ack_range.start(), *ack_range.end())),
```

Comment:
If we know the range type that is going to be consumed from the iter, why not just have the iter return PacketNumberRange?